NixOS 21.05: evdi fails to build with latest nixpkgs update (displaylink dependency)

### Describe the bug

A related chunk of log of `nixos-rebuild`:

```
/build/source/module/evdi_modeset.c:331:19: error: initialization of 'void (*)(struct drm_plane *, struct drm_atomic_state *)' from incompatible pointer type 'void (*)(struct drm_plane *, struct drm_plane_state *)' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
  331 |  .atomic_update = evdi_plane_atomic_update,
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~
/build/source/module/evdi_modeset.c:331:19: note: (near initialization for 'evdi_plane_helper_funcs.atomic_update')
/build/source/module/evdi_modeset.c:332:16: error: 'drm_gem_fb_prepare_fb' undeclared here (not in a function); did you mean 'drm_gem_fb_create'?
  332 |  .prepare_fb = drm_gem_fb_prepare_fb
      |                ^~~~~~~~~~~~~~~~~~~~~
      |                drm_gem_fb_create
/build/source/module/evdi_modeset.c:336:19: error: initialization of 'void (*)(struct drm_plane *, struct drm_atomic_state *)' from incompatible pointer type 'void (*)(struct drm_plane *, struct drm_plane_state *)' [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Werror=incompatible-pointer-types8;;]
  336 |  .atomic_update = evdi_cursor_atomic_update,
      |                   ^~~~~~~~~~~~~~~~~~~~~~~~~
/build/source/module/evdi_modeset.c:336:19: note: (near initialization for 'evdi_cursor_helper_funcs.atomic_update')
cc1: all warnings being treated as errors
make[4]: *** [/nix/store/zbc1cq2y4dkgfhxjmv7q5441v0jviyrb-linux-5.13.6-dev/lib/modules/5.13.6/source/scripts/Makefile.build:273: /build/source/module/evdi_modeset.o] Error 1
make[3]: *** [/nix/store/zbc1cq2y4dkgfhxjmv7q5441v0jviyrb-linux-5.13.6-dev/lib/modules/5.13.6/source/Makefile:1853: /build/source/module] Error 2
make[2]: *** [/nix/store/zbc1cq2y4dkgfhxjmv7q5441v0jviyrb-linux-5.13.6-dev/lib/modules/5.13.6/source/Makefile:220: __sub-make] Error 2
make[2]: Leaving directory '/nix/store/zbc1cq2y4dkgfhxjmv7q5441v0jviyrb-linux-5.13.6-dev/lib/modules/5.13.6/build'
make[1]: *** [Makefile:77: module] Error 2
make[1]: Leaving directory '/build/source/module'
make: *** [Makefile:8: all] Error 2
```

### Steps To Reproduce
Steps to reproduce the behavior:
1. Update `nixpkgs` channel to https://releases.nixos.org/nixos/21.05/nixos-21.05.1970.11c662074e2 of 31 July 2021 (previous working state is https://releases.nixos.org/nixos/21.05/nixos-21.05.1523.b2f87e0043a of 17 July 2021)
2. Run `sudo nixos-rebuild boot`

### Expected behavior
NixOS rebuilds successfully, DisplayLink works.

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->
- `linuxPackages.evdi` @eyJhb 
- `displaylink` @nshalman @abbradar @peterhoeg @eyJhb

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.12.15, NixOS, 21.05.1523.b2f87e0043a (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.12`
 - channels(root): `"nixos, nixos-unstable"`
 - channels(wenzel): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- linuxPackages.evdi
- displaylink
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/os-specific/linux/evdi/default.nix
+++ b/pkgs/os-specific/linux/evdi/default.nix
@@ -1,28 +1,15 @@
-{ lib, stdenv, fetchFromGitHub, fetchpatch, kernel, libdrm }:
+{ lib, stdenv, fetchFromGitHub, kernel, libdrm }:
 
 stdenv.mkDerivation rec {
   pname = "evdi";
-  version = "unstable-2021-06-11";
+  version = "unstable-2021-07-07";
 
   src = fetchFromGitHub {
     owner = "DisplayLink";
     repo = pname;
-    rev = "65e12fca334f2f42396f4e8d16592d53cab34dd6";
-    sha256 = "sha256-81IfdYKadKT7vRdkmxzfGo4KHa4UJ8uJ0K6djQCr22U=";
+    rev = "b0b2c80eb63f9b858b71afa772135f434aea192a";
+    sha256 = "sha256-io+CbZovGjEJjwtmARFH23Djt933ONoHMDoea+i6xFo=";
   };
-
-  # Linux 5.13 support
-  # The patches break compilation for older kernels
-  patches = lib.optional (kernel.kernelAtLeast "5.13") [
-    (fetchpatch {
-      url = "https://github.com/DisplayLink/evdi/commit/c5f5441d0a115d2cfc8125b8bafaa05b2edc7938.patch";
-      sha256 = "sha256-tWYgBrRh3mXPebhUygOvJ07V87g9JU66hREriACfEVI=";
-    })
-    (fetchpatch {
-      url = "https://github.com/DisplayLink/evdi/commit/5f04d2e2df4cfd21dc15d31f1152c6a66fa48a78.patch";
-      sha256 = "sha256-690/eUiEVWvnT/YAVgKcLo86dgolF9giWRuPxXpL+eQ=";
-    })
-  ];
 
   nativeBuildInputs = kernel.moduleBuildDependencies;
 

